demos/src/iot_demo_mqtt.c

```diff
@@ -417,6 +418,21 @@ static int _establishMqttConnection( bool awsIotMqttMode,
     connectInfo.keepAliveSeconds = KEEP_ALIVE_SECONDS;
     connectInfo.pWillInfo = &willInfo;
 
+    if( pCredentials != NULL )
+    {
+        connectInfo.pUserName = pCredentials->pUserName;
+        connectInfo.userNameLength = ( uint16_t ) pCredentials->userNameSize;
+        connectInfo.pPassword = pCredentials->pPassword;
+        connectInfo.passwordLength = ( uint16_t ) pCredentials->passwordSize;
+    }
+    else
+    {
+        connectInfo.pUserName = NULL;
+        connectInfo.userNameLength = 0;
+        connectInfo.pPassword = NULL;
+        connectInfo.passwordLength = 0;
+    }
+
```

Comment:
Is this something that should be added to `_establishMqttConnection()` in the other AWS demos (shadow, jobs, defender), or just for this one?

Reply:
Good question. My understanding is that the MQTT CLI demo - plus the library itself - is sufficient for the near-term deadline.